hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/service/SnapshotDeletingService.java

```diff
@@ -147,12 +148,12 @@ public BackgroundTaskResult call() throws InterruptedException {
 
         while (iterator.hasNext() && snapshotLimit > 0) {
           SnapshotInfo snapInfo = iterator.next().getValue();
-          SnapshotInfo.SnapshotStatus snapshotStatus =
-              snapInfo.getSnapshotStatus();
+          boolean isSstFilteringServiceEnabled =
+              ((KeyManagerImpl) ozoneManager.getKeyManager())
+                  .isSstFilteringSvcEnabled();
 
           // Only Iterate in deleted snapshot
-          if (!snapshotStatus.equals(
-              SnapshotInfo.SnapshotStatus.SNAPSHOT_DELETED)) {
+          if (shouldIgnoreSnapshot(snapInfo, isSstFilteringServiceEnabled)) {
```

Comment:
Curious if we need to have test for it in `TestSnapshotDeletingService` as well or [this](https://github.com/apache/ozone/pull/5155/files#diff-c4c691f6dfa73df833ed29be86d77ef32b206052a5c69febef2c39eaef5fcc51R256) is enough?.

Reply:
Will add this in a follow up jira.